perfact/zodbsync/zodbsync.py

```diff
@@ -55,74 +56,69 @@ def mod_format(data=None, indent=0, as_list=False):
+                quote = "'"
+                repl.append(("'", "\\'"))
+            for orig, r in repl:
+                val = val.replace(orig, r)
 
-    def make_line(line):
-        output.append(indent * ' ' + line)
+            return ("u" if is_unicode else "") + quote + val + quote
+        else:
+            return str((val,))[1:-2]
 
     # Convert dictionary to sorted list of tuples (diff-friendly!)
     if isinstance(data, dict):
         data = [(key, value) for key, value in data.items()]
         data.sort()
 
-    make_line('[')
-    indent += 4
-    for item in data:
-        if isinstance(item[1], list):
-            # Non-trivial lists are shown on separate lines.
-            lines = item[1]
-            if len(lines) > 1:
-                make_line("("+str_repr(item[0])+", [")
-                indent += 4
-                for l in lines:
-                    make_line(str_repr(l) + ',')
-                make_line("]),")
-                indent -= 4
-            else:
-                make_line(str(item)+',')
-
-        elif isinstance(item[1], (bytes, unicode)):
-            # Multiline presentation of non-trivial text / blobs
-            text = item[1]
-            if isinstance(text, bytes):
-                newline = b'\n'
-            else:
-                newline = u'\n'
-            if text != '' and (text.find(newline) != -1 or len(text) > 80):
-                # Keep newlines after splitting.
-                lines = conserv_split(text, newline)
-                # Could be binary data. So, split superlong lines as well.
-                lines = split_longlines(lines)
-
-                make_line("("+str_repr(item[0])+", ")
-                indent += 4
-                for l in lines:
-                    make_line(str_repr(l) + '+')
-                make_line("''),")
-                indent -= 4
-            else:
-                make_line(str(item)+',')
+    # The data is now given by a list of tuples, each of which has two elements
+    # (diff-friendly version of a dict). The first element of the tuple is a
+    # string, while the second one might be any combination of lists, tuples
+    # and PODs (unicode, bytes, numbers, booleans ...). Usually, we keep each
+    # element in one line. An exception are lists with multiple elements, which
+    # allow an additional indentation, being split over multiple lines.
+    output = []
+    output.append('[')
+    for key, value in data:
+        key_repr = '    (%s, ' % str_repr(key)
+        if isinstance(value, list) and len(value) > 1:
+            # Non-trivial lists are split onto separate lines.
+            output.append(key_repr + '[')
+            for item in value:
+                output.append('        %s,' % str_repr(item))
+            output.append('        ]),')
         else:
-            make_line(str(item)+',')
-    indent -= 4
-    make_line(']')
+            output.append(key_repr + '%s),' % str_repr(value))
+    output.append(']')
```

Comment:
This is a lot of work... Can't we use `pprint` and be done with it? (I know, the formatting is different, just saying...)

Reply:
I'm afraid pprint will also use the bytes representation:

    In [22]: print(pprint.saferepr('testü'))
    'test\xc3\xbc'
    In [23]: pprint.pprint('testü')
    'test\xc3\xbc'

Reply:
I was afraid it might.

Your proposal is really the best way to handle this.